api/src/middlewares/services/UserService/personalisation/personalisationBuilder/userPostalConfirmation/getAdditionalPersonalisations.ts

```diff
@@ -0,0 +1,33 @@
+import { AnswersHashMap } from "../../../../../../types/AnswersHashMap";
+
+export type AdditionalPersonalisationCountries = "Spain" | "Italy" | "Croatia";
+
+type PersonalisationFunction = (fields: AnswersHashMap) => Record<string, boolean>;
+
+export const postalPersonalisationTypeMap: Record<AdditionalPersonalisationCountries, PersonalisationFunction> = {
```

Comment:
I don't think we're using a `type` to map anything here? (i.e. we're not mapping by a type, we're mapping by country)

 `postalPersonalisationByCountry`

and you'd use it like

```
const getPostalPersonalisation = postalPersonalisationByCountry["country"]

const additionalPersonalisations = getPostalPersonalisation?.(answers)
```

doing a spread operator on undefined/null shouldn't have any affect, does ts complain otherwise?